tool/internal_pkg/pluginmode/thriftgo/hessian2.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package thriftgo
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/cloudwego/kitex/tool/internal_pkg/generator"
+	"github.com/cloudwego/kitex/tool/internal_pkg/log"
+	"github.com/cloudwego/kitex/tool/internal_pkg/util"
+	"github.com/cloudwego/thriftgo/config"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	JavaExtensionOption = "java_extension"
+
+	DubboCodec = "github.com/kitex-contrib/codec-dubbo"
+	JavaThrift = "https://raw.githubusercontent.com/kitex-contrib/codec-dubbo/main/java/java.thrift"
+)
+
+// Hessian2PreHook Hook before building cmd
+func Hessian2PreHook(cfg *generator.Config) {
+	// add thrift option
+	cfg.ThriftOptions = append(cfg.ThriftOptions, "template=slim,with_reflection,code_ref")
+
+	// run hessian2 options
+	for _, opt := range cfg.Hessian2Options {
+		runOption(cfg, opt)
+	}
+}
+
+// runOption Execute the corresponding function according to the option
+func runOption(cfg *generator.Config, opt string) {
+	switch opt {
+	case JavaExtensionOption:
+		runJavaExtensionOption(cfg)
+	}
+}
+
+// runJavaExtensionOption Pull the extension file of java class from remote
+func runJavaExtensionOption(cfg *generator.Config) {
+	// get java.thrift
+	if p := util.JoinPath(filepath.Dir(cfg.IDL), "java.thrift"); !util.Exists(p) {
+		if err := util.DownloadFile(JavaThrift, p); err != nil {
+			log.Warn("Downloading java.thrift file failed:", err.Error())
+			os.Exit(1)
+		}
+	}
+
+	// load / create the idl-ref config file
```

Comment:
Refactor line 66~89 to a standalone function `patchIDLRefConfig()` to make it easier to understand.